src/components/Assistant.vue

```diff
@@ -0,0 +1,431 @@
+	},
+	beforeDestroy() {
+		if (!this.showAssistant) {
+			return
+		}
+
+		this.$editor.off('selectionUpdate', this.onSelection)
+		clearInterval(this.$interval)
+	},
+	methods: {
+		async fetchTasks() {
+			const result = await axios.get(generateOcsUrl('/textprocessing/tasks/app/text') + '?identifier=' + this.identifier)
+
+			const taskMap = {}
+			for (const index in this.providers) {
+				const provider = this.providers[index]
+				taskMap[provider.task] = provider
+			}
+
+			this.tasks = result.data.ocs.data.tasks.map((task) => {
+				return {
+					...task,
+					title: taskMap[task.type].name,
+				}
+			}).sort((a, b) => b.id - a.id)
+		},
+		onSelection() {
+			const { state } = this.$editor
+			const { from, to } = state.selection
+			this.selection = state.doc.textBetween(from, to, ' ')
+		},
+		async openAssistantForm(taskType = null) {
+			await window.OCA.TPAssistant.openAssistantForm(
+				{
+					appId: 'text',
+					identifier: this.identifier,
+					taskType,
+					input: this.selection,
+					isInsideViewer: true,
+				})
+			await this.fetchTasks()
+		},
+		openTranslateDialog() {
+			this.displayTranslate = this.selection
+		},
+		hideTranslate() {
+			this.displayTranslate = false
+		},
+		translateInsert(content) {
+			this.$editor.commands.command(({ tr, commands }) => {
+				return commands.insertContentAt(tr.selection.to, content)
+			})
+			this.displayTranslate = false
+		},
+		translateReplace(content) {
+			this.$editor.commands.command(({ tr, commands }) => {
+				const selection = tr.selection
+				const range = {
+					from: selection.from,
+					to: selection.to,
+				}
+				return commands.insertContentAt(range, content)
+			})
+			this.displayTranslate = false
+		},
+		async openResult(task) {
+			window.OCA?.TPAssistant.openAssistantResult(task)
+		},
+		async insertResult(task) {
+			this.$editor.commands.insertContent(task.output)
+			this.showTaskList = false
+		},
+		async copyResult(task) {
+			try {
+				await navigator.clipboard.writeText(task.output)
+				showSuccess(t('text', 'Nextcloud Assistant result copied'))
+				this.showTaskList = false
+			} catch (error) {
+				console.error(error)
+				showError(t('text', 'Could not copy result to clipboard'))
+			}
+		},
+		async deleteTask(task) {
+			try {
+				await axios.delete(generateOcsUrl(`/textprocessing/task/${task.id}`))
+			} catch (e) {
+				console.error('Failed to delete task', e)
+			}
+			const taskIndex = this.tasks.findIndex(t => t.id === task.id)
+			if (taskIndex > -1) {
+				this.$delete(this.tasks, taskIndex)
+			}
+		},
+		floatingOptions() {
+			const buttonSize = 44
+			const topSpacing = 110 + 22
+			const bottomSpacing = 50 + 22
+			return {
+				placement: 'right',
+				getReferenceClientRect: () => {
+					const editorRect = this.$parent.$el.querySelector('.ProseMirror').getBoundingClientRect()
+					const pos = posToDOMRect(this.$editor.view, this.$editor.state.selection.from, this.$editor.state.selection.to)
+					return {
```

Comment:
Just leaving this here in case relevant for future debugging. Adding a div to position above the editor for fiddling with the positioning has been quite helpful:

```
// Add template
<div class="floating-hover" />

// Add styles
.floating-hover {
	background-color: rgba(255,0,0,0.4);
	position: fixed;
	top:0;
	left: 0;
	width: 100px;
	height: 100px;
}

// Add here
document.querySelector('.floating-hover').style.width = rect.width + 'px'
document.querySelector('.floating-hover').style.height = rect.height + 'px'
document.querySelector('.floating-hover').style.top = rect.top + 'px'
document.querySelector('.floating-hover').style.left = rect.left + 'px'
document.querySelector('.floating-hover').style.bottom = rect.bottom + 'px'
document.querySelector('.floating-hover').style.right = rect.right + 'px'
```